### ISSUE TITLE
Java Support

### ISSUE THREAD

**Issue by redknightlois** [Microsoft/bond#103]
Would be great to have support for Java. Today I has to help design an application with an Android component and the lack of Java prevented us to use Bond. Would have been great because mobile is a great place for efficient & compact serialization methods.

**Comment by ardanew** [comment]
upvote

**Comment by TikhonV** [comment]
+1

**Comment by salmangadit** [comment]
+1

**Comment by chadwalters** [comment]
Yes, we are definitely interested in supporting Java. If anyone is interested in working on this, please get in touch.

**Comment by jprobichaud** [comment]
+1  This would enable some interesting work on the android side

**Comment by jeffsink** [comment]
I see there is some awesome progress on Java here, are we getting close?

**Comment by tstein** [comment]
We've got a good chunk of the core functionality done. You should be able to generate code from any bond file (except for `service` definitions) and use it for serialization/deserialization in any of our binary protocols.

There's no release yet, and we have to satisfy some internal requirements before we can provide binaries, but if you're interested and don't mind a little legwork, I would encourage you to build bond for yourself. You can do this by following our existing build guide and then building the `java` target (`cmake --build . --target java` or `make java`). If you run into any issues, please let us know, and we'll be happy to fix them.

**Comment by hcoona** [comment]
Also want Java version bond-over-gRPC
